### PR TITLE
[AC 1526]Show current SM seat and service account usage in Bitwarden Portal

### PR DESCRIPTION
**Pull request by cyprain-okeke**
Screenshot 2023-07-31 at 23 34 26
Screenshot 2023-07-31 at 23 34 38
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This ensures the org is billed correctly and the customer has a seamless experience transitioning to a paid sub.

We should display:

Occupied SM seats

Number of projects

Total number of service accounts in use  

in the Organization page in Bitwarden Portal.

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Admin/Controllers/OrganizationsController.cs
Comment (eliykat) [inline src/Admin/Controllers/OrganizationsController.cs:187]: As a follow-up, I think we should combine all these DB calls into 1 query and model that contains all the required information. It's now 12 separate queries just for 1 controller endpoint. What do you think? If you agree, please create a follow-up ticket and link it here.

File: src/Admin/Models/OrganizationViewModel.cs
Comment (eliykat) [inline src/Admin/Models/OrganizationViewModel.cs:45]: Seems a bit weird that we're now storing numbers as strings here. I think it's better to have this logic in the template, e.g.: https://github.com/bitwarden/server/blob/d35de15dfa444ec0af56f3027b3f4ff49f41c96b/src/Admin/Views/Organizations/_ViewInformation.cshtml#L20-L22

so that these properties can just be `int`s.

File: src/Admin/Startup.cs
Comment (eliykat) [inline src/Admin/Startup.cs:12]: This needs to be moved within the `#if !OSS` block (~line 15 below). That ensures that it's only included in the compilation for non-OSS builds.

File: src/Admin/Admin.csproj
Comment (eliykat) [inline src/Admin/Admin.csproj:10]: Need to move down within the conditional block, line 22

**Review by eliykat — CHANGES_REQUESTED** [review]
Nice work, can you also share a screenshot showing the UI change?

**Comment by eliykat** [comment]
When @bitwarden/team-secrets-manager-dev review: do you want these repo methods returning the objects, or just the count, given that we only need the count? (i.e. `IEnumerable<Secret>` or `int`?)

File: src/Admin/Controllers/OrganizationsController.cs
Comment (eliykat) [inline src/Admin/Controllers/OrganizationsController.cs:154]: We also need to show the occupied SM seats. It looks like this is missing from the Figma, so you may need to check back with Sukhleen.

File: src/Admin/Controllers/OrganizationsController.cs
Comment (cyprain-okeke) [reply src/Admin/Controllers/OrganizationsController.cs:187]: I agree with you. I have created the follow-up [ticket](https://bitwarden.atlassian.net/browse/AC-1569) and link it

File: src/Admin/Models/OrganizationViewModel.cs
Comment (cyprain-okeke) [reply src/Admin/Models/OrganizationViewModel.cs:45]: resolved

File: src/Admin/Startup.cs
Comment (cyprain-okeke) [reply src/Admin/Startup.cs:12]: done

File: src/Admin/Admin.csproj
Comment (cyprain-okeke) [reply src/Admin/Admin.csproj:10]: done

File: src/Admin/Controllers/OrganizationsController.cs
Comment (Thomas-Avery) [inline src/Admin/Controllers/OrganizationsController.cs:74]: For the OSS build/running, won't this cause a runtime error when going to the admin portal, since the secret and project repos don't have a Noop implementation?

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
A few things to check. 

Also, I would recommend if all we need is a count of these things let's use the same approach as we did with service accounts. https://github.com/bitwarden/server/blob/f69bf1adb4359bccb2c3b777cd671c4353f28941/bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/ServiceAccountRepository.cs#L133-L141

File: bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs
Comment (cyprain-okeke) [reply bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs:65]: Thanks for this insight, I will check this with Max

File: src/Admin/Controllers/OrganizationsController.cs
Comment (cyprain-okeke) [reply src/Admin/Controllers/OrganizationsController.cs:154]: I have added the secrets manager seats count while awaits Sukhleen to Update the UI

File: src/Admin/Views/Organizations/_ViewInformation.cshtml
Comment (eliykat) [inline src/Admin/Views/Organizations/_ViewInformation.cshtml:45]: Indentation needs fixing

File: src/Admin/Controllers/OrganizationsController.cs
Comment (eliykat) [reply src/Admin/Controllers/OrganizationsController.cs:72]: Yes I think it will, great catch. We can just crank out some more noop implementations, although if you have a better idea I would love to hear it. We were talking [above](https://github.com/bitwarden/server/pull/3142#discussion_r1278655937) about combining these repo calls later, is there any way we could reasonably combine these SM-related calls into 1, then we only need 1 noop repo?

File: src/Admin/Views/Organizations/_ViewInformation.cshtml
Comment (eliykat) [inline src/Admin/Views/Organizations/_ViewInformation.cshtml:46]: Secrets manager isn't released for self-hosted yet, so we should either hide these fields or set to N/A.

We'll need to come back and update this check when SM is released for self-hosted. (CC @Thomas-Avery )

Comment (cyprain-okeke) [reply src/Admin/Views/Organizations/_ViewInformation.cshtml:46]: I think using Model.UseSecretsManager to check should solve the problem

File: src/Admin/Views/Organizations/_ViewInformation.cshtml
Comment (cyprain-okeke) [reply src/Admin/Views/Organizations/_ViewInformation.cshtml:45]: fixied

File: src/Admin/Controllers/OrganizationsController.cs
Comment (Thomas-Avery) [reply src/Admin/Controllers/OrganizationsController.cs:72]: I would recommend for now just swapping them to fetching <int> and using noop.

Then, when the work on AC-1569 starts, refactor to fetching a DTO via an org repository that has all the information we will need fetched from the DB.

File: src/Admin/Models/OrganizationViewModel.cs
Comment (eliykat) [inline src/Admin/Models/OrganizationViewModel.cs:67]: You can access this via the existing `Organization` property without adding a new property. You can refer to it via a getter for convenience if you want though.

Comment (cyprain-okeke) [reply src/Admin/Models/OrganizationViewModel.cs]: resolved

**Review by eliykat — APPROVED** [review]
LGTM but please wait for @Thomas-Avery 's approval as well,  thanks!

File: src/Core/SecretsManager/Repositories/Noop/NoopServiceAccountRepository.cs
Comment (Thomas-Avery) [inline src/Core/SecretsManager/Repositories/Noop/NoopServiceAccountRepository.cs:60]: I don't think we need this anymore?

We have `GetManyByOrganizationIdAsync(Guid organizationId, Guid userId, AccessClientType accessType)` at line 8 which covers the interface.

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
One minor thing, everything else looks good!

**Review by Thomas-Avery — APPROVED** [review]
SM changes look good, thank you

**Pull request merged**
